File: 0344-reverse-string/0344-reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        self.solution(s, 0, len(s)-1)
    
    def solution(self, s, l, r):
        if l >= r:
            return
        s[l], s[r] = s[r], s[l]
        self.solution(s, l + 1, r - 1)

File: 0344-reverse-string/0344-reverse-string.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
    s.reverse()
    return s
    
    for (let i = 0; i < s.length / 2; i++) {
        let [first, last] = [s[i], s[s.length - i - 1]];
        s[i] = last;
        s[s.length - i - 1] = first;
    }
    return s.join("");
};